examples/general.c

```diff
@@ -708,10 +709,32 @@ static void config_files(const char *repo_path, git_repository* repo)
 	sprintf(config_path, "%s/config", repo_path);
 	check_error(git_config_open_ondisk(&cfg, config_path), "opening config");
 
-	git_config_get_int32(&j, cfg, "help.autocorrect");
-	printf("Autocorrect: %d\n", j);
+	error_code = git_config_get_int32(&j, cfg, "help.autocorrect");
+	switch (error_code)
+	{
+		case 0:
+			printf("Autocorrect: %d\n", j);
+			break;
+		case GIT_ENOTFOUND:
+			printf("Autocorrect: Undefined\n");
+			break;
+		default:
+			check_error(error_code, "get_int32 failed");
+	}
+	git_config_free(cfg);
 
 	check_error(git_repository_config_snapshot(&snap_cfg, repo), "config snapshot");
-	git_config_get_string(&email, snap_cfg, "user.email");
-	printf("Email: %s\n", email);
+	error_code = git_config_get_string(&email, snap_cfg, "user.email");
+	switch (error_code)
```

Comment:
I did cut and paste tge fragments starting with the "switch".  A macro would be a nice way to avoid the duplication, but it will have to take a number of parameters:
 1) The displayed name of the variable (Autocorrect or Email).
 2) The variable's type (%d or %s).
 3) The value of the variable.
 4) The error message for check_error().
 5) Maybe "error_code" so the user of the macro could use any name for this variable.

I'll be glad to add a macro if you think that's nicer.